common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -198,6 +198,8 @@ def delete_item(self, location, user_id, revision=None, skip_auto_publish=False,
                 branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
             elif revision is None:
                 branches_to_delete = [ModuleStoreEnum.BranchName.draft]
+                if location.category in DIRECT_ONLY_CATEGORIES:
+                    branches_to_delete.append(ModuleStoreEnum.BranchName.published)
```

Comment:
All about modules (including `video`) are actually just re-named `HTMLModules`. The video gets special treatment from studio to parse a youtube id from the html block, and then write the HTML snippet back into the course when an edit happens.

[Updating an about item](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/models/settings/course_details.py#L101-L121)
[Building the video about item html](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/models/settings/course_details.py#L200-L201)
[Video item parsing and serialization](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/models/settings/course_details.py#L207-L235)

So, we just have to make sure that we correctly delete the about item (which is a single XBlock) from the course tree when we remove it, which is what this change does.